framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
```

Comment:
It's vestigial from the original issue which only deals with registered projects.